src/app/fdctl/run/tiles/fd_shred.c

```diff
@@ -342,35 +357,58 @@ during_frag( void * _ctx,
     return;
   }
 
-  if( FD_UNLIKELY( in_idx!=NET_IN_IDX ) ) {
-    /* This is a frag from the bank tile. We can just go ahead and shred
-       it here, even though we may get overrun.  If we do end up getting
-       overrun, we just won't send these shreds out and we'll reuse the
-       FEC set for the next one.  From a higher level though, if we do
-       get overrun, a bunch of shreds will never be transmitted, and
-       we'll end up producing a block that never lands on chain. */
+  if( FD_UNLIKELY( in_idx==POH_IN_IDX ) ) {
+    /* This is a frag from the bank tile.  We'll copy it to our pending
+       microblock batch and shred it if necessary (last in block or
+       above watermark).  We just go ahead and shred it here, even
+       though we may get overrun.  If we do end up getting overrun, we
+       just won't send these shreds out and we'll reuse the FEC set for
+       the next one.  From a higher level though, if we do get overrun,
+       a bunch of shreds will never be transmitted, and we'll end up
+       producing a block that never lands on chain. */
     fd_fec_set_t * out = ctx->fec_sets + ctx->shredder_fec_set_idx;
 
     uchar const * dcache_entry = fd_chunk_to_laddr_const( ctx->poh_in_mem, chunk );
-    if( FD_UNLIKELY( chunk<ctx->poh_in_chunk0 || chunk>ctx->poh_in_wmark ) || sz>FD_BANK_SHRED_MTU )
+    if( FD_UNLIKELY( chunk<ctx->poh_in_chunk0 || chunk>ctx->poh_in_wmark ) || sz>FD_POH_SHRED_MTU )
       FD_LOG_ERR(( "chunk %lu %lu corrupt, not in range [%lu,%lu]", chunk, sz,
             ctx->poh_in_chunk0, ctx->poh_in_wmark ));
 
-    fd_entry_batch_meta_t const * entry_batch_meta = (fd_entry_batch_meta_t const *)dcache_entry;
-    uchar const *                 entry_batch      = dcache_entry + sizeof(fd_entry_batch_meta_t);
-    ulong                         entry_batch_sz   = sz           - sizeof(fd_entry_batch_meta_t);
+    fd_entry_batch_meta_t const * entry_meta = (fd_entry_batch_meta_t const *)dcache_entry;
+    uchar const *                 entry      = dcache_entry + sizeof(fd_entry_batch_meta_t);
+    ulong                         entry_sz   = sz           - sizeof(fd_entry_batch_meta_t);
+
+    /* It should never be possible for this to fail, but we check it
+       anyway. */
+    FD_TEST( entry_sz + ctx->pending_batch.pos <= sizeof(ctx->pending_batch.payload) );
+    FD_TEST( (ctx->pending_batch.microblock_cnt==0) | (ctx->pending_batch.slot==entry_meta->slot) );
+
+    ctx->pending_batch.slot = entry_meta->slot;
+    /* Ugh, yet another memcpy */
+    fd_memcpy( ctx->pending_batch.payload + ctx->pending_batch.pos, entry, entry_sz );
+    ctx->pending_batch.pos += entry_sz;
+    ctx->pending_batch.microblock_cnt++;
+
+    int last_in_batch = (entry_meta->tick==entry_meta->bank_max_tick_height) | (ctx->pending_batch.pos > PENDING_BATCH_WMARK);
+
+    if( FD_UNLIKELY( last_in_batch )) {
+      fd_shredder_init_batch( ctx->shredder, ctx->pending_batch.raw, ctx->pending_batch.pos, entry_meta );
```

Comment:
```suggestion
      fd_shredder_init_batch( ctx->shredder, ctx->pending_batch.raw, ctx->pending_batch.pos + sizeof(ctx->pending_batch.microblock_cnt), entry_meta );
```
I think we need this +8 here, but I'm not sure why my tests worked fine.